src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,681 @@
+    protected boolean loadACHBatchDetailFile(String inputFileName, BatchInputFileType batchInputFileType) {
+        boolean result = true;
+        
+        byte[] fileByteContent = safelyLoadFileBytes(inputFileName);
+        
+        LOG.info("Attempting to parse the file.");
+        
+        Object parsedObject = null;
+        try {
+            parsedObject = batchInputFileService.parse(batchInputFileType, fileByteContent);
+        } catch (ParseException e) {
+            String errorMessage = "Error parsing batch file: " + e.getMessage();
+            LOG.error(errorMessage, e);
+            throw new RuntimeException(errorMessage);
+        }
+        
+        if (!(parsedObject instanceof List)) {
+            String errorMessage = "Parsed file was not of the expected type.  Expected [" + List.class + "] but got [" + parsedObject.getClass() + "].";
+            criticalError(errorMessage);
+        }
+        
+        @SuppressWarnings("unchecked")
+        List<PayeeACHAccountExtractDetail> achDetails = (List<PayeeACHAccountExtractDetail>) parsedObject;
+        
+        for (PayeeACHAccountExtractDetail achDetail : achDetails) {
+            if (!processACHBatchDetail(achDetail)) {
+                result = false;
+            }
+        }
+        
+        return result;
+    }
+
+    /**
+     * Processes a single ACH batch detail, and routes a Payee ACH Account maintenance document accordingly.
+     */
+    protected boolean processACHBatchDetail(PayeeACHAccountExtractDetail achDetail) {
+        Person payee = personService.getPersonByPrincipalName(achDetail.getNetID());
+        
+        if (!validateACHBatchDetail(achDetail, payee)) {
+            return false;
+        }
+        
+        // Check for existing ACH accounts.
+        PayeeACHAccount entityAccount = achService.getAchInformation(
+                PayeeIdTypeCodes.ENTITY, payee.getEntityId(), getDirectDepositTransactionType());
+        PayeeACHAccount employeeAccount = achService.getAchInformation(
+                PayeeIdTypeCodes.EMPLOYEE, payee.getEmployeeId(), getDirectDepositTransactionType());
+        
+        // Add or update Entity ID account.
+        if (ObjectUtils.isNull(entityAccount)) {
+            addACHAccount(payee, achDetail, PayeeIdTypeCodes.ENTITY);
+            LOG.info("Created new ACH Account of Entity type for payee " + payee.getPrincipalName());
+        } else {
+            updateACHAccountIfNecessary(payee, achDetail, entityAccount);
+        }
+        
+        // Add or update Employee ID account.
+        if (ObjectUtils.isNull(employeeAccount)) {
+            addACHAccount(payee, achDetail, PayeeIdTypeCodes.EMPLOYEE);
+            LOG.info("Created new ACH Account of Employee type for payee " + payee.getPrincipalName());
+        } else {
+            updateACHAccountIfNecessary(payee, achDetail, employeeAccount);
+        }
+        
+        return true;
+    }
+

```

Comment:
The update action may not actually perform an update (even if we did intend to implement the update feature now), but I will see if I can log some sort of message to indicate an attempt was made.